rs/src/management/tunnel_request_options.rs

```diff
@@ -17,19 +17,33 @@ pub struct TunnelRequestOptions {
     /// Gets or sets a flag that requests tunnel ports when retrieving a tunnel object.
     pub include_ports: bool,
 
+    /// Gets or sets an optional list of tags to filter the requested tunnels or ports.
+    pub tags: Vec<String>,
+
+    /// Gets or sets a flag that indicates whether listed items must match all tags
+    /// specified in `tags`. If false, an item is included if any tag matches.
+    pub require_all_tags: bool,
```

Comment:
As a small tweak I might make this an

```
pub enum SearchTags {
    Any(Vec<String>),
    All(Vec<String>),
}
```

since `require_all_tags` is a no-op without `tags` being specified.

Note this would require a manual implementation of Default, like

```
impl Default for SearchTags {
   pub fn default() -> Self {
        SearchTags::Any(vec![])
    }
}
```

Reply:
Interesting suggestion. I often don't recognize all the applications for Rust's more powerful enums. But in this case it seems like it adds a bit of complexity (additional enum type and necessity of referencing it to specify tags) without bringing much value. And it makes the Rust tunnel API model a little more different from other SDK languages, which (cumulatively) will add to maintenance overhead. So for those reasons I'll decline the suggestion.